lib/checks/hasUnsupportedNodes.js

```diff
@@ -1,30 +1,29 @@
-const Node = require('../node');
+const {query} = require('math-nodes');
+
 const resolvesToConstant = require('./resolvesToConstant');
 
 function hasUnsupportedNodes(node) {
-  if (Node.Type.isParenthesis(node)) {
-    return hasUnsupportedNodes(node.content);
-  }
-  else if (Node.Type.isUnaryMinus(node)) {
-    return hasUnsupportedNodes(node.args[0]);
-  }
-  else if (Node.Type.isOperator(node)) {
-    return node.args.some(hasUnsupportedNodes);
-  }
-  else if (Node.Type.isSymbol(node) || Node.Type.isConstant(node)) {
-    return false;
-  }
-  else if (Node.Type.isFunction(node, 'abs')) {
-    if (node.args.length !== 1) {
-      return true;
+  if (query.isApply(node)) {
+    if (query.isAbs(node)) {
+      if (node.args.length !== 1) {
+        return true;
+      }
+      if (hasUnsupportedNodes(node.args[0])) {
```

Comment:
since all the `Apply` nodes are caught in the same if branch and all check `hasUnsupportedNodes`, we could just check that in the beginning for all of them